enhancements/oc/oc-login-external-oidc-integration.md

```diff
@@ -0,0 +1,228 @@
+
+There are currently 7 authentication flows grouped around public and confidential clients. Due to the nature of `oc` that is being used widely on local, 
+we have decided to support authentication flows only for public clients for now but if there is any different use case in the future, this command can support confidential clients too. 
+
+Supportability (in terms of authentication flows) matrix of this feature is outlined below; 
+
+|           | Authorization Code<br/>(public/confidential)                        | Authorization Code + PKCE<br/>(public/confidential)                 | Implicit Flow<br/>(public) | Client Credentials<br/>(confidential)                                    | Refresh Token<br/>(public/confidential) | Device Code<br/>(public) | Password Grant<br/>(confidential) |
+|-----------|---------------------------------------------------------------------|---------------------------------------------------------------------|----------------------------|--------------------------------------------------------------------------|-----------------------------------------|--------------------------|-----------------------------------|
+| Supported | Yes                                                                 | Yes                                                                 | No                         | No                                                                       | Yes                                     | No                       | No                                |
+| Comment   | Can be used for confidential clients<br/>by passing --client-secret | Can be used for confidential clients<br/>by passing --client-secret | Deprecated                 | Mostly used in CI systems that <br/>client-secret can be stored securely | If provider supports it                 |                          | Deprecated                        |
+
+
+### API Extensions
+
+N/A
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+### Risks and Mitigations
+
+There is additional configuration steps should be performed in external OIDC side which may have slight negative impact on usability initially.
+
+### Drawbacks
+
+There is no drawback because this feature has no relation with the current functionality.
+
+## Open Questions [optional]
+
+
+## Test Plan
+
+Currently, test plan is mostly managed around unit test but in the future it is inevitable to have e2e test executed by our CI system.
+
+## Graduation Criteria
+
+N/A
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+N/A
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+N/A
+
+## Version Skew Strategy
+
+Command will fail gracefully during authentication against older clusters or the clusters external OIDC is not enabled.
```

Comment:
add sample output of what happens when 
1. the copy login command is provided to old `oc`
2. an old `oc` is used with a kubeconfig created by new `oc`

Reply:
Added.